### PR TITLE
Add nfs-ha-cluster-ubuntu

### PR DESCRIPTION
**Pull request by hosungsmsft**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Initial nfs-ha-cluster-ubuntu check-in

### CODE REVIEW

File: nfs-ha-cluster-ubuntu/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+![cluster_diagram](images/NFS-HA-Arch.png "Diagram of deployed cluster")
+
+## Usage
+
+### Template Parameters
+
+There are many parameters this template requires. Some notable and important ones are as follows:
+
+- `subnetId`: This is where the highly available NFS cluster will be deployed. You need to give the Azure resource ID of an already existing Azure subnet where the cluster should be deployed. You may find that the template accept `default` as this parameter's value, but that's only for the Azure quickstart templates repo's CI validation purpose (because for CI, we can't provide an already existing vnet, so we have to create one on the fly, and `default` will do just that---however, the IP ranges are fixed to `10.0.0.0/16` for the vnet and `10.0.0.0/24` for the subnet, so you shouldn't really rely on this unless your desired IP ranges match these hard-coded ones).
```
Comment (bmoore-msft) [inline nfs-ha-cluster-ubuntu/README.md]: See: https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#template-pre-requisites for how to use prereqs in CI

File: nfs-ha-cluster-ubuntu/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                        "value": "[parameters('_artifactsLocation')]"
+                    },
+                    "_artifactsLocationSasToken": {
+                        "value": "[parameters('_artifactsLocationSasToken')]"
+                    },
+                    "location": {
+                        "value": "[parameters('location')]"
+                    },
+                    "subnetId": {
+                        "value": "[if(equals(parameters('subnetId'), 'create-for-ci'), reference('nfs-ha-vnet-default-deployment').outputs.subnetId.value, parameters('subnetId'))]"
```
Comment (bmoore-msft) [inline nfs-ha-cluster-ubuntu/azuredeploy.json]: let's remove this pattern and rely on the prereqs for creating and existing resources that are needed - that will simplify for customer that want to adopt this

File: nfs-ha-cluster-ubuntu/README.md (hunk 1 above)
Comment (hosungsmsft) [reply nfs-ha-cluster-ubuntu/README.md]: Thanks for the pointer. Updated the template w/ the suggested GEN-PREREQ practice.

File: nfs-ha-cluster-ubuntu/azuredeploy.json (hunk 2 above)
Comment (hosungsmsft) [reply nfs-ha-cluster-ubuntu/azuredeploy.json]: Done as commented. Thanks.

**Pull request merged**